### PR TITLE
Clean all those safe_div, _safe_div methods

### PR DESCRIPTION
**Pull request by facaiy**
tf.div_no_nan was introduced by #21621, which is designed to replace all those safe_div methods here.

Ref: #21784

~~@martinwicke please add a API design label, because we add a new argument: negative_to_zero.~~

### CODE REVIEW

File: tensorflow/contrib/losses/python/losses/loss_ops.py [hunk 1]
```diff
@@ -104,7 +78,8 @@ def _safe_mean(losses, num_present):
       then zero is returned.
   """
   total_loss = math_ops.reduce_sum(losses)
-  return _safe_div(total_loss, num_present)
+  return math_ops.div_no_nan(total_loss, num_present,
+                             negative_to_zero=True, name="value")
```
Comment (martinwicke) [inline tensorflow/contrib/losses/python/losses/loss_ops.py]: I'd much prefer to replace `div_no_nan(x, y, negative_to_zero=True, ...)` with `div_no_nan(x, tf.max(y, 0), ...)` and not add the `negative_to_zero` argument, which I doesn't look to me like it makes things simpler.

Comment (facaiy) [reply tensorflow/contrib/losses/python/losses/loss_ops.py]: Reasonable. I'd like to revise code as suggested later.

**Comment by facaiy** [comment]
@martinwicke  negative_to_zero was removed.

File: tensorflow/contrib/losses/python/losses/loss_ops.py [hunk 2]
```diff
@@ -104,7 +78,9 @@ def _safe_mean(losses, num_present):
       then zero is returned.
   """
   total_loss = math_ops.reduce_sum(losses)
-  return _safe_div(total_loss, num_present)
+  return math_ops.div_no_nan(total_loss,
+                             math_ops.maximum(num_present, 0),
```
Comment (facaiy) [inline tensorflow/contrib/losses/python/losses/loss_ops.py]: I think `maximum` is unnecessary op here, because num_present should be non-negative in practice.

**Comment by facaiy** [comment]
Hi, do you have any update here?

**Comment by facaiy** [comment]
Hey, what am I doing wrong? It seems that both reviewers are selected by github automatically.

File: tensorflow/contrib/losses/python/losses/loss_ops.py (hunk 2 above)
Comment (alextp) [reply tensorflow/contrib/losses/python/losses/loss_ops.py]: Does something break when you remove the maximum? This could be another case of us needing the double-tap-where trick to avoid NaNs in the gradient.

Comment (facaiy) [reply tensorflow/contrib/losses/python/losses/loss_ops.py]: Hi, I don't remove the maximum. Test failed after I resolved the conflict with the lastest master. I'll tell you when I fix it later. :-)

**Review by alextp — APPROVED** [review]

**Comment by facaiy** [comment]
Hi, any update here? thanks
